pkg/network/usm/monitor_test.go

```diff
@@ -808,3 +814,360 @@ func TestCleanProtocolMaps(t *testing.T) {
+	redisBuffer  = []byte("+AAA\r\n")
+	randomBuffer = []byte("random payload")
+	amqpBuffer   = []byte("AMQP")
+)
+
+func TestConnectionStatesMap(t *testing.T) {
+	skipTestIfKernelNotSupported(t)
+
+	httpEnabledConfig := withConfigChange(utils.NewUSMEmptyConfig(), func(cfg *networkConfig.Config) { cfg.EnableHTTPMonitoring = true })
+	http2EnabledConfig := withConfigChange(utils.NewUSMEmptyConfig(), func(cfg *networkConfig.Config) { cfg.EnableHTTP2Monitoring = true })
+	postgresEnabledConfig := withConfigChange(utils.NewUSMEmptyConfig(), func(cfg *networkConfig.Config) { cfg.EnablePostgresMonitoring = true })
+	redisEnabledConfig := withConfigChange(utils.NewUSMEmptyConfig(), func(cfg *networkConfig.Config) { cfg.EnableRedisMonitoring = true })
+	kafkaEnabledConfig := withConfigChange(utils.NewUSMEmptyConfig(), func(cfg *networkConfig.Config) { cfg.EnableKafkaMonitoring = true })
+
+	tests := make([]connectionStatesMapTestCase, 0)
+	tests = append(tests, connectionStatesMapTestCase{
+		name:                "HTTP protocol enabled",
+		cfg:                 httpEnabledConfig,
+		expectedResult:      shouldExists,
+		sendRequestCallback: sendAndReadBuffer(httpBuffer),
+	}, connectionStatesMapTestCase{
+		name:                "HTTP protocol disabled",
+		cfg:                 http2EnabledConfig, // Enabling any protocol other than HTTP to allow USM to run
+		expectedResult:      shouldNotExists,
+		sendRequestCallback: sendAndReadBuffer(httpBuffer),
+		skipCondition:       skipIfKernelNotSupported,
+	}, connectionStatesMapTestCase{
+		name:           "HTTP protocol already classified",
+		cfg:            httpEnabledConfig,
+		expectedResult: shouldExists,
+		preTestSetup:   markConnectionProtocol(protocols.HTTP),
+	}, connectionStatesMapTestCase{
+		name:           "HTTP protocol already classified but not enabled",
+		cfg:            redisEnabledConfig, // Enabling any protocol other than HTTP to allow USM to run
+		expectedResult: shouldNotExists,
+		preTestSetup:   markConnectionProtocol(protocols.HTTP),
+	}, connectionStatesMapTestCase{
+		name:                "HTTP2 protocol enabled",
+		cfg:                 http2EnabledConfig,
+		expectedResult:      shouldExists,
+		sendRequestCallback: sendAndReadBuffer([]byte(http2.ClientPreface)),
+		skipCondition:       skipIfKernelNotSupported,
+	}, connectionStatesMapTestCase{
+		name:                "HTTP2 protocol disabled",
+		cfg:                 httpEnabledConfig, // Enabling any protocol other than HTTP2 to allow USM to run
+		expectedResult:      shouldNotExists,
+		sendRequestCallback: sendAndReadBuffer([]byte(http2.ClientPreface)),
+	}, connectionStatesMapTestCase{
+		name:           "HTTP2 protocol already classified",
+		cfg:            http2EnabledConfig,
+		expectedResult: shouldExists,
+		preTestSetup:   markConnectionProtocol(protocols.HTTP2),
+		skipCondition:  skipIfKernelNotSupported,
+	}, connectionStatesMapTestCase{
+		name:           "HTTP2 protocol already classified but not enabled",
```

Comment:
what about gRPC, the classification is not the same as HTTP/2?

Reply:
to classify grpc we first need to classify http2